codex/raw/marketplace.md

```diff
@@ -0,0 +1,275 @@
+---
+title: CODEX-MARKETPLACE
+name: Codex Storage Marketplace
+status: raw
+tags: codex
+editor: Dmitriy <dryajov@status.im>
+contributors:
+- Mark <mark@codex.storage>
+- Adam <adam.u@status.im>
+- Eric <ericmastro@status.im>
+- Jimmy Debe <jimmy@status.im>
+---
+
+## Abstract
+
+This specification describes a method for Codex storage providers and client nodes to participate in a storage marketplace. 
+The goal is to create a storage marketplace that promotes durability.
+
+## Motivation
+The Codex network aims to create a peer-to-peer storage engine with strong data durability, 
+data persistence guarantees and node storage incentives.
+Support for resource restricted devices, like mobile devices should also be embraced.
+The protocol should remove complexity to allow for a simple implementation and 
+simplify incentive mechanisms.
+
+## Semantics 
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+| Terminology  | Description |
+| --------------- | --------- |
+| storage providers | A Codex node that provides storage services to the marketplace. |
+| validator nodes | A Codex node that checks for missing storage proofs and triggers for a reward. |
+| client nodes | The most common Codex node that interacts with other nodes to store, locate and retrieve data. |
+| slots | Created by client nodes when a new dataset is requested to be stored. Discussed further in the [slots section](#slots).  |
+
+### Storage Request
```

Comment:
I am a bit confused here, this section only defines the values of a storage request. Also validators do not really need a section as they are only mentioned a few times. This section could be changed to Client Storage Request.  A validator and storage node roles should be explained in a different rfc how they interact with the marketplace. Also, maybe the name of this rfc should be changed to Client Marketplace and a separate marketplace smart contract rfc should be created to defined the all aspects of the smart contract , which you mentioned before in a sync meeting.

Reply:
@jimstir well, then I am confused as well 😅

As you accepted my edit to the abstract, which clearly states that this spec defines interactions with the Marketplace smart contract for all the different roles in the network, including Clients, Storage Providers, and Validators (as they all interact with the smart contract)... So I would have expected to have those included here. Is your plan to have separate specs for each of them? I am not sure if that is a good idea because a lot of things overlap, and hence you will have to link between the specs a lot, which to me does not really sound good. I was actually expecting to have all those definitions in this spec, which in the end is called "Marketplace." Is this spec then focusing on Clients? If so, then this spec should be renamed, but again I would advise against that.

Honestly, Validator RFC does not really make sense to me that much as it is such a simple functionality that can be described in one or two paragraphs. If you want, we can jump on a call and I can explain its functionality very quickly.

> Also, a separate marketplace smart contract rfc should be created to defined the all aspects of the smart contract

Hmm, I don't really remember what I was saying about it back then, but now I don't think we need the spec for that. For whom would it be? For what purpose? I think all the needed information about the contract should be placed in this spec.

Reply:
@AuHau I do think there can be sections briefly explaining how storage providers and validators interact with the contract. For the most part, I believe their interactions are included throughout the spec. For clarity purposes, the other roles should be detailed in separate specs and reference here. Validators and storage providers have components that do not particularly relate to the marketplace itself, like the data storage mechanism, messaging method, resource requirements, etc. 

> Hmm, I don't really remember what I was saying about it back then, but now I don't think we need the spec for that. For whom would it be? For what purpose? I think all the needed information about the contract should be placed in this spec.

Right, I believe we came to that conclusion during the sync meeting also, after I asked you. Sorry I forgot for a second.